spec/github_helper_spec.rb

```diff
@@ -217,31 +217,77 @@ def get_milestone(milestone_name:)
+      due_date = '2022-06-18T23:00:00Z'.to_time
+      options = {
+        due_on: '2022-06-19T12:00:00Z',
+        description: "Code freeze: June 19, 2022\nApp Store submission: June 21, 2022\nRelease: June 22, 2022\n"
+      }
 
       expect(client).to receive(:create_milestone).with(test_repo, test_milestone_number, options)
-      create_milestone(need_submission: true, milestone_duration: 19, days_code_freeze: 3)
+      create_milestone(due_date: due_date, days_until_submission: 2, days_until_release: 3)
     end
 
-    def create_milestone(need_submission:, milestone_duration:, days_code_freeze:)
-      days_until_submission = need_submission ? (days_code_freeze - 3) : milestone_duration
+    it 'computes the correct dates when the due date is on DST but has no day change' do
+      # The PST to PDT (DST) change is made on the second Sunday in March and finishes on the first Sunday in November
+      Time.zone = 'Pacific Time (US & Canada)'
+      due_date = '2022-06-18T22:00:00Z'.to_time
+      options = {
+        due_on: '2022-06-18T12:00:00Z',
+        description: "Code freeze: June 18, 2022\nApp Store submission: June 20, 2022\nRelease: June 21, 2022\n"
+      }
+
+      expect(client).to receive(:create_milestone).with(test_repo, test_milestone_number, options)
+      create_milestone(due_date: due_date, days_until_submission: 2, days_until_release: 3)
+    end
+
+    context 'with input validation' do
+      it 'raises an error if days_until_submission is less than or equal zero' do
+        due_date = '2022-10-20T08:00:00Z'.to_time.utc
+        expect { create_milestone(due_date: due_date, days_until_submission: 0, days_until_release: 5) }
+          .to raise_error(FastlaneCore::Interface::FastlaneError, 'days_until_submission must be greater than zero.')
+      end
+
+      it 'raises an error if days_until_release is less than or equal zero' do
+        due_date = '2022-10-20T08:00:00Z'.to_time.utc
+        expect { create_milestone(due_date: due_date, days_until_submission: 12, days_until_release: -8) }
+          .to raise_error(FastlaneCore::Interface::FastlaneError, 'days_until_release must be greater than zero.')
+      end
+
+      it 'raises an error if submission date is after the release date' do
+        due_date = '2022-10-20T08:00:00Z'.to_time.utc
+        expect { create_milestone(due_date: due_date, days_until_submission: 14, days_until_release: 3) }
+          .to raise_error(FastlaneCore::Interface::FastlaneError, 'days_until_release must be greather than days_until_submission')
+      end
+
+      it 'raises an error if submission date is equal to release date' do
+        due_date = '2022-10-20T08:00:00Z'.to_time.utc
+        expect { create_milestone(due_date: due_date, days_until_submission: 1, days_until_release: 1) }
+          .to raise_error(FastlaneCore::Interface::FastlaneError, 'days_until_release must be greather than days_until_submission')
+      end
```

Comment:
This should be a valid case imho (see other comment)

Reply:
Yeah, I was a bit confused if this should be or not be a valid case, Initially, I think it was, but as [you explained here](https://github.com/wordpress-mobile/release-toolkit/pull/426#discussion_r1016860739) about the process, it feels that the submissions always need to be before release, and not fall in the same day. I'll change that test to be a valid case, thanks for the input. 😄